api/nodes/serializers.py

```diff
@@ -494,6 +497,11 @@ def get_current_user_can_comment(self, obj):
         else:
             return obj.can_comment(auth)
 
+    def get_current_user_is_contributor(self, obj):
+        user = self.context['request'].user
+        user = None if user.is_anonymous else user
+        return obj.is_contributor(user)
```

Comment:
Worried about adding n+1 queries here. Are there annotations you could take advantage of? From `get_current_user_permissions`, it looks like you might be able to check `contrib_admin`, `contrib_write`, `contrib_read`.

Reply:
Yeah, I think that makes sense.